### PR TITLE
Add grid-size support to style property.

### PR DESCRIPTION
**Pull request by jeresig**
Add in support for passing a function to the `style` property of our HOC so that we can display certain styles based on the current grid size. This should replace the need for most media queries in our code base (we'll just want to wrap our site in one grid as a base but we can always override it further down).

### CODE REVIEW

**Comment by jeresig** [comment]
Deploy preview for *wonder-blocks* ready!

Built with commit 0b53d096ee428045172ed3c90818a81a1877aae2

https://deploy-preview-82--wonder-blocks.netlify.com

File: packages/wonder-blocks-core/generated-snapshot.test.js [hunk 1]
```diff
@@ -56,16 +56,10 @@ describe("wonder-blocks-core", () => {
...
-
-        const example = (
-            <StyledInput
-                style={styles.pink}
-                type="text"
-                placeholder="hello, world"
-            />
-        );
+        
+        const example = <StyledInput style={styles.pink} type="text" placeholder="hello, world"/>;
```
Comment (kevinbarabash) [inline packages/wonder-blocks-core/generated-snapshot.test.js]: Why the change in formatting?

File: packages/wonder-blocks-core/generated-snapshot.test.js [hunk 2]
```diff
@@ -56,16 +56,10 @@ describe("wonder-blocks-core", () => {
                 backgroundColor: "pink",
             },
         });
-
+
```
Comment (kevinbarabash) [inline packages/wonder-blocks-core/generated-snapshot.test.js]: We shouldn't have extra whitespace.

File: packages/wonder-blocks-core/generated-snapshot.test.js (hunk 1 above)
Comment (kevinbarabash) [reply packages/wonder-blocks-core/generated-snapshot.test.js]: Can you run `npm run pretty-quick` manually?

File: packages/wonder-blocks-core/util/util.js [hunk 3]
```diff
@@ -3,14 +3,20 @@ import {css} from "aphrodite";
...
 import type {StyleType} from "./types.js";
 
-function flatten<T: Object>(list?: StyleType<T>): Array<T> {
+export type GridSize = "small" | "medium" | "large";
+
+function flatten<T: Object>(list?: StyleType<T>, gridSize: GridSize): Array<T> {
     const result: Array<T> = [];
 
+    if (typeof list === "function") {
+        list = list(gridSize);
```
Comment (kevinbarabash) [inline packages/wonder-blocks-core/util/util.js]: This makes sense.

File: packages/wonder-blocks-core/generated-snapshot.test.js (hunk 1 above)
Comment (jeresig) [reply packages/wonder-blocks-core/generated-snapshot.test.js]: pretty-quick doesn't do anything if there are no changes. I ended up having to run `./node_modules/.bin/prettier --write packages/wonder-blocks-*/generated-snapshot.test.js` - but that changed even more files (I think prettier isn't getting run against these files for some reason).

**Comment by jeresig** [comment]
@kevinbarabash I fixed the formatting issue, should be good-to-go.

File: package.json [hunk 4]
```diff
@@ -7,7 +7,7 @@
     "install": "for d in packages/*/ ; do (cd \"$d\" && npm install); done",
     "lint": "eslint --config ./eslint/eslintrc packages",
     "flow": "flow",
-    "gen-snapshot-tests": "node utils/gen-snapshot-tests.js",
+    "gen-snapshot-tests": "node utils/gen-snapshot-tests.js && prettier --write packages/*/generated-snapshot.test.js",
```
Comment (jeresig) [inline package.json]: @kevinbarabash I added this in and now files are being properly formatted all the time!

**Comment by kevinbarabash** [comment]
@jeresig thanks for fixing the prettier issue.
